packages/twenty-server/src/core/auth/services/auth.service.ts

```diff
@@ -191,6 +195,10 @@ export class AuthService {
   }
 
   async checkUserExists(email: string): Promise<UserExists> {
+    if (this.environmentService.isSignUpDisabled()) {
```

Comment:
I was about to say that this looks a bit hacky but I actually think it makes sense to prevent a malicious user to list users.

Reply:
That was actually an advice from @FelixMalfait so kudos to him! 🙌